codegenerator/cli/templates/static/codegen/src/eventFetching/ChainFetcher.res

```diff
@@ -283,6 +283,12 @@ let hasProcessedToEndblock = (self: t) => {
   }
 }
 
+let hasNoMoreEventsToProcess = (self: t, ~hasArbQueueEvents) => {
+  !hasArbQueueEvents &&
+  !self.isFetchingBatch &&
+  self.fetchState->PartitionedFetchState.queueSize === 0
```

Comment:
Also, started calculating the queue size only when other conditions return true

Reply:
Nice